kubetest/azure.go

```diff
@@ -101,12 +111,93 @@ type Cluster struct {
 	acsCustomWinBinariesURL string
 	acsEngineBinaryPath     string
 	acsCustomCcmURL         string
+	azureEnvironment        string
+	azureIdentitySystem     string
+	azureCustomCloudURL     string
 	agentPoolCount          int
 	k8sVersion              string
 	networkPlugin           string
 	azureClient             *AzureClient
 }
 
+// IsAzureStackCloud return true if the cloud is AzureStack
+func (c *Cluster) isAzureStackCloud() bool {
+	return c.azureCustomCloudURL != "" && strings.EqualFold(c.azureEnvironment, AzureStackCloud)
+}
+
+// SetCustomCloudProfileEnvironment retrieves the endpoints from Azure Stack metadata endpoint and sets the values for azure.Environment
+func (c *Cluster) SetCustomCloudProfileEnvironment() error {
+	var environmentJSON string
+	if c.isAzureStackCloud() {
+
+		env := azure.Environment{}
+		env.Name = c.azureEnvironment
+		azsFQDNSuffix := strings.Replace(c.azureCustomCloudURL, fmt.Sprintf("https://portal.%s.", c.location), "", -1)
+		azsFQDNSuffix = strings.TrimSuffix(azsFQDNSuffix, "/")
+		env.ResourceManagerEndpoint = fmt.Sprintf("https://management.%s.%s/", c.location, azsFQDNSuffix)
+		metadataURL := fmt.Sprintf("%s/metadata/endpoints?api-version=1.0", strings.TrimSuffix(env.ResourceManagerEndpoint, "/"))
```

Comment:
would this work if the kubetest is running on non-Azure environments?

Reply:
Yes. I tested changes and after above changes Kubetest will be able to launched on Azure Stack env.